### ISSUE TITLE
Enhancement: Calcite-icon option for adornments

### ISSUE THREAD

**Issue by TheBlueDog** [Esri/calcite-components#1767]
### Description
Enhance the `calcite-icon` component by adding the option to display an adornment layer over the chosen icon.

There are cases where an icon needs additional definition, but is not interactive like a button.

Icon library will not require additional versions of every icon to be created each time the adornment is needed by a product.

### Acceptance Criteria 
- I can choose to display an adornment on an icon by setting an option in the component
- I can specify which adornment from a defined list to display
- I can specify the color variable to use for a fill on the adornment
- The adornments can be applied to all of the icon size options
- The adornments will always be displayed in the lower right corner of the component for LTR locales
- The adornments will always be displayed in the lower left corner of the component when RTL option is set

### Relevant Info 
This will bring a familiar pattern from Pro into the web platform. See: ArcGIS Pro Style Guide [Icon Adornments](http://prodesignguide.esri.com/documentation/control-elements/icon-adornments/)

**Dependency** - The adornments will need to be created by the icon team. It is unknown where the adornments would be stored (part of Calcite UI Icon library?)

#### Which Component
[Calcite Icon](https://esri.github.io/calcite-components/?path=/story/components-icon--simple)

#### Example Use Case

Implementation code suggestion:
```
<calcite-icon icon="file-xls" scale="m" adornment="expired" adornment-fill="ui-danger"></calcite-icon>
```
![Screen Shot 2021-03-16 at 8 56 20 PM](https://user-images.githubusercontent.com/5419306/111399521-ed0c6f00-869b-11eb-977d-48084949d5c3.png)

**Comment by dozer76** [comment]
This will be an awesome addition for Site Scan's upcoming feature - Enterprise Fleet Management. We need to use icons to represent certain actions like this in interactive tables, which can be thought of as activity feeds. examples include: 

- Maintenance event was completed on a drone
- Maintenance event is past-due
- A new document has been added to the organization (i.e. training guide)
- A document has expired, or renewed (i.e. pilot certificate)
- An incident occurred during the flight

**Comment by johnmgriffith** [comment]
@macandcheese I wanted to tackle those points, because I had some the same thoughts. You brought up some great questions.

- For the smaller icons and larger icons, they all use the same size adornment, using an 11px circle. You can see in Erich's screenshot that all the adornments are the same size, and since these are primarily for statuses I think that's the way to go. This would be especially viable if we could scoot the adornment off the viewbox a couple pixels.

- We'd need to work out what is the best icon for those elements. For example, "File has expired" would work better with a clock instead of a 'sync' icon. The meanings of the icons combined with the colors should make it clearer.

- I think the adornments should be reserved for statuses primarily. Actionable items should be baked in as part of the icon, grey, and not contained in a circle, just like the icon `folder-plus`

- I believe we should clarify that adornments do not belong on icons that already have modifiers. That would definitely cause some confusion. Perhaps we can pick out which icons have modifiers, and not permit them to use the adornment function?

- The trouble with just colored dot indicators is that it doesn't convey _enough_ meaning. Also there's accessibility problems when relying on color to convey all the meaning. Adornment + color container can indicate a lot more information than the dot.

**Comment by johnmgriffith** [comment]
> * Some of the above already have calcite-ui-icons that could just use a fill color? (layer-add, layer-broken, check-layer, lock, etc.) - by adding more iconography on top of any underlying icon - are we making is easy for folks to create confusion for their users

Yeah, also, `layer-broken` and `check-layer` could be removed if we have these adornments implemented. We'd just have to make sure that people who are using those icons aren't using them as actionable icons. Otherwise we should keep them as is.

**Comment by paulcpederson** [comment]
@johnmgriffith do we have these glyphs available? Assuming you would want these drawn to fit the 11x11 artboard right?

Could also have a `calcite-adornment` component which you could pass as the default slot to an icon. This would enable you to use an adornment with other components as well like `calcite-avatar` or `calcite-action`

**Comment by macandcheese** [comment]
@paulcpederson @jul11557 -
We are looking at this again and are wondering:
- Is being able to display an icon a requirement, or would the ability to set a "status dot" on top of Avatar be sufficient?

**Comment by macandcheese** [comment]
@Esri/calcite-pes @Esri/calcite-designers propose we close this.

**Comment by ashetland** [comment]
👍

**Comment by geospatialem** [comment]
Closing the above out as unplanned due to the enhancement request, timing and other priorities.
